akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1146,19 +1146,15 @@ class DDataShardCoordinator(
         context.become(waitingForUpdate(evt, afterUpdateCallback, newRemainingKeys))
 
     case UpdateTimeout(CoordinatorStateKey, Some(`evt`)) =>
-      if (terminated) {
-        log.error(
-          "The ShardCoordinator was unable to update a distributed state within 'updating-state-timeout': {} millis (terminating). " +
-          "Perhaps the ShardRegion has not started on all active nodes yet? event={}",
-          writeMajority.timeout.toMillis,
-          evt)
-        onTerminated()
+      log.error(
+        "The ShardCoordinator was unable to update a distributed state within 'updating-state-timeout': {} millis ({}). " +
+        "Perhaps the ShardRegion has not started on all active nodes yet? event={}",
+        writeMajority.timeout.toMillis,
+        if (terminating) "terminating" else "retrying",
+        evt)
+      if (terminating) {
+        onTerminating()
```

Comment:
my thoughts about stop vs `onTerminating` was for all the UpdateTimeout and ModifyFailure cases, also here I think it should stop

Reply:
You are right.